plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -246,6 +249,23 @@ protected void refreshFromMetadataLocation(String newLocation)
         shouldRefresh = false;
     }
 
+    private boolean shouldBeAborted(Throwable failure)
+    {
+        if (failure instanceof org.apache.iceberg.exceptions.NotFoundException) {
+            return true;
+        }
+        if (failure.getCause() != null) {
+            Throwable cause = failure.getCause();
+            if (cause instanceof FileNotFoundException
+                    || cause instanceof TrinoS3FileSystem.UnrecoverableS3OperationException
```

Comment:
`TrinoS3FileSystem.UnrecoverableS3OperationException` we shouldn't do such low-level checks here.

Reply:
This low-level handling is purely based on what `TrinoS3FileSystem` throws when `newStream` is called.

`FileNotFoundException` or `UnrecoverableS3OperationException` are `IOException` which get caught and wrapped with `RuntimeException` in `TableMetadataParser#read`.

That is why I had to use `failure.getCause()`. If we do not abort on `UnrecoverableS3OperationException` exception, Then it will retry the operation 20 times which is costly.

Reply:
Removed this code and handle `FileNotFoundException` in [/ForwardingSeekableInputStream.java](https://github.com/trinodb/trino/pull/16674/commits/cef5d099878d9007a5d4dba859ad28d8db138148#diff-d2efd26a616b31a9e87eab43321c46a74a89d6ff4ad21104fbc5023cba4ca56f)

However, It is still needed to handle `TrinoS3FileSystem.UnrecoverableS3OperationException` exception.

Reply:
Can you pls create a follow-up issue for this kind of situation?

Reply:
https://github.com/trinodb/trino/issues/16908